pkgs/esp-idf/default.nix

```diff
@@ -120,6 +120,12 @@ stdenv.mkDerivation rec {
   dontUseNinjaCheck = true;
 
   installPhase = ''
+    # make esp-idf cmake git version detection happy
+    git init .
+    git config user.email "nixbld@localhost"
+    git config user.name "nixbld"
+    git commit --date="1970-01-01 00:00:00" --allow-empty -m "make idf happy"
```

Comment:
This seems quite clever.  Is it working on your end?

Reply:
I'm getting this error now:

```fatal: detected dubious ownership in repository at '/nix/store/mfmmywqch5m4bshhmh5lb01np9yjdlgk-esp-idf-v5.1.2'
To add an exception for this directory, call:

        git config --global --add safe.directory /nix/store/mfmmywqch5m4bshhmh5lb01np9yjdlgk-esp-idf-v5.1.2```

Reply:
I tried fiddling with this for a few hours but didn't crack it.  You can see the things I tried here:

https://github.com/jonahbron/nixpkgs-esp-dev/commits/madmo/?author=jonahbron

Hoping you have more luck!